pkg/api/v1/atlasdeployment_types.go

```diff
@@ -250,6 +250,25 @@ type AutoScalingSpec struct {
 	Compute *ComputeSpec `json:"compute,omitempty"`
 }
 
+// AdvancedAutoScalingSpec configures your deployment to automatically scale its storage
+type AdvancedAutoScalingSpec struct {
+	// Flag that indicates whether autopilot mode for Performance Advisor is enabled.
+	// The default is false.
+	AutoIndexingEnabled *bool `json:"autoIndexingEnabled,omitempty"`
+	// Flag that indicates whether disk auto-scaling is enabled. The default is true.
+	// +optional
+	DiskGBEnabled *DiskGBEnabled `json:"diskGB,omitempty"`
```

Comment:
I think we can rename the field to `DiskGB` to match json object.

Also potentially can rename `DiskGBEnabled` struct to something general like `Enabled` so it can be reused somewhere (even thought there is nowhere to reuse it for now).

Reply:
Renamed to DiskGB.
Second option is too general. Reusing fields with common names was the cause of this bug.

Reply:
Thanks for the change :)

In the first part I was actually talking about the `DiskGBEnabled` field itself, not the type.
Can agree with you regarding the potential change 👍